app/models/manageiq/providers/microsoft/infra_manager/refresh_parser.rb

```diff
@@ -330,6 +330,8 @@ def process_vm_hardware(vm)
     end
 
     def process_snapshots(p)
+      return unless p[:VMCheckpoints]
```

Comment:
Hrm, so now it's a question of semantics...

The `save_inventory` code treats an empty array differently from `nil`.

An empty array means: Delete all existing snapshots for this VM in our inventory.

`nil` means: skip processing snapshots for this VM.

So, if this VM previously had snapshots, returning `nil` here means "keep the old snapshots in our inventory for this VM".  Whereas, returning `[]` here means "delete the old snapshots in our inventory for this VM".

I don't know which one is correct based on this strange value for inventory.

Granted, this looks like it fixes the error in the bug.  But, I don't know if it ends up with the right result in our inventory.  And, I have no suggestions for figuring that out ... seems like a strange result from SCVMM.